src/controllers/socialController.js

```diff
@@ -0,0 +1,102 @@
+import jwt from 'jsonwebtoken';
+import {
+  config
+} from 'dotenv';
+import db from '../models';
+
+config();
+
+class SocialController {
+  static async findOrCreateUserByEmail(firstName, lastName, email) {
+    const [{
+      dataValues: userData
+    }] = await db.User.findOrCreate({
+      defaults: {
+        firstName,
+        lastName,
+        email
+      },
+      where: {
+        email
+      }
+    });
+    return userData;
+  }
+
+  static async googleLogin(req, res) {
+    // check if user exist
+    try {
+      const {
+        _json: user,
+        provider
+      } = req.user;
+      if (!user.email) {
+        return res.status(401).json({
+          status: 'false',
+          error: 'No email address is found for this user'
```

Comment:
we use a `message` property to send a response back to the user. also, it should be a `404`, not a `401`